src/Modules/IexecPoco.ts

```diff
@@ -421,3 +423,11 @@ export function handleFaultyContribution(event: FaultyContributionEvent): void {
     workerAccount.score = faultyContributionEvent.score;
     workerAccount.save();
 }
+export function handleDealSponsored(event: DealSponsoredEvent): void {
+    let dealSponsoredEvent = new DealSponsored(createEventID(event));
+    dealSponsoredEvent.transaction = logTransaction(event).id;
```

Comment:
you should load an entity before creating it. It prevent bugs when the graphNode make some re-indexing behavior: 

```suggestion
    let dealSponsored = DealSponsored.load(createEventID(event))
    if (!dealSponsored){
            dealSponsored = new DealSponsored(createEventID(event));
    }
     dealSponsored.transaction = logTransaction(event).id;
```